wgpu-types/src/lib.rs

```diff
@@ -122,6 +122,25 @@ pub struct RequestAdapterOptions<S> {
     pub compatible_surface: Option<S>,
 }
 
+impl<S> RequestAdapterOptions<S> {
+    pub fn new() -> Self {
+        Self {
+            power_preference: PowerPreference::default(),
+            compatible_surface: None,
+        }
+    }
+
+    pub fn power_preference(&mut self, preference: PowerPreference) -> &mut Self {
```

Comment:
I generally think of builders as taking `mut self` as opposed to `&mut self` so that the final value is movable. @kvark what do you think?

Reply:
I do somewhat agree. That would mean that the return value should be `Self`. @cwfitzgerald was there a particular reason for passing the value by reference over moving the value around?

Reply:
The reasoning behind using `&mut` builders is that it is more flexible when doing complex init, and because all our descriptors are accepted by reference, it can be passed in as is with no issues.

Reply:
Let's proceed with `&mut self` like originally done by @cwfitzgerald . If users want to move things out, they can always `clone()`, but I imagine this is going to be a minority case.